.github/workflows/e2e-tests.yml

```diff
@@ -53,86 +84,77 @@ jobs:
     - name: Setup tools
       uses: ./.github/actions/tool-setup
 
-    - name: Build Production Jetpack
-      env:
-        COMPOSER_ROOT_VERSION: "dev-master"
+    - name: Install monorepo
       run: |
         pnpm install
-        pnpx jetpack build packages/search -v --production
-        pnpx jetpack build plugins/jetpack -v --production
 
-    - name: Environment set-up
-      working-directory: projects/plugins/jetpack/tests/e2e
+    - name: Test environment set-up
+      working-directory: ${{ matrix.path }}
       env:
         CONFIG_KEY: ${{ secrets.E2E_CONFIG_KEY }}
+        COMPOSER_ROOT_VERSION: "dev-master"
+        RUN_NAME: ${{ needs.create-test-matrix.outputs.runName }}
       run: |
+        echo "::group::Install e2e project"
         pnpm install
-        pnpm run test-decrypt-all-config
-        pnpm run env-start
-        pnpm run tunnel-on
+        echo "::endgroup::"
+
+        echo "::group::Build plugin(s)"
+        pnpm run build
+        echo "::endgroup::"
+
+        echo "::group::Decrypt config"
+        pnpm run config:decrypt
+        echo "::endgroup::"
+
+        echo "::group::Start docker environment"
+        pnpm run env:up
+        echo "::endgroup::"
+
+        if [ "${RUN_NAME}" == gutenberg ]; then
+            echo "::group::Setting up Gutenberg"
+            pnpx e2e-env gb-setup
+            echo "::endgroup::"
+        fi
+
+        echo "::group::Create tunnel"
+        pnpm run tunnel:up
+        echo "::endgroup::"
 
-    - name: Set up Gutenberg
-      if: matrix.group == 'gutenberg'
-      working-directory: projects/plugins/jetpack/tests/e2e
-      run: pnpx e2e-env gb-setup
+    - name: Run ${{ matrix.project }} tests
+      working-directory: ${{ matrix.path }}
+      env:
+        TEST_ARGS: ${{ matrix.args }}
+      run: pnpm run test:run -- "${ARGS}"
```

Comment:
Two problems on this line.

1. That should probably be `TEST_ARGS`, not `ARGS`. Or rename the variable on line 127.
2. As written, for the Atomic test it looks like it'll pass a single arg "blocks --grep-invert wordads", rather than three args "blocks", "--grep-invert", and "wordads".

For the second one, as long as none of the args themselves will need quoting then you should be safe just doing `$TEST_ARGS` instead of `"$TEST_ARGS"`. If they might need quoting, you'd have to do something fancier, maybe have it be a JSON array and do something like this
```bash
mapfile -t ARGS < <(jq -r '.[]' <<<"$JSON_ARGS")
pnpm run test:run -- "${ARGS[@]}"
```

Reply:
I ended up using a JSON array for both test args and slack args.